pyanaconda/screen_access.py

```diff
@@ -0,0 +1,226 @@
+#
+# screen_access.py:  anaconda thread management
+#
+# Copyright (C) 2016
+# Red Hat, Inc.  All rights reserved.
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+# Author(s):  Martin Kolman <mkolman@redhat.com>
+#
+import os
+CONFIG_FILE_NAME = "anaconda"
+CONFIG_FILE_PATH = os.path.join("/etc/sysconfig", CONFIG_FILE_NAME)
+CONFIG_TRUE = "1"
+CONFIG_FALSE = "0"
+CONFIG_BOOLEAN_TABLE = {
+    CONFIG_TRUE: True,
+    CONFIG_FALSE: False
+}
+CONFIG_VISITED_KEY = "visited"
+CONFIG_GENERAL_SECTION = "General"
+CONFIG_DISABLE_POSTINST_TOOLS_KEY = "post_install_tools_disabled"
+CONFIG_OPTION_PREFIX = "changed_"
+
+import logging
+log = logging.getLogger("anaconda")
+
+from ConfigParser import ConfigParser
+from threading import RLock
+
+from pyanaconda import startup_utils
+from pyanaconda import iutil
+
+
+class ScreenAccessManager(object):
+    """A singleton that takes care about spoke visits and option changes.
+
+    The visits and option changes are reflected in the user interaction config file.
+    For specification of the user interaction config file see the
+    user-interaction-config-file-spec.rst file in the docs directory of the
+    Anaconda source code checkout.
+    """
+
+    def __init__(self):
+        self._lock = RLock()
+        self._config = ConfigParser()
+
+    def open_config_file(self, config_path=CONFIG_FILE_PATH):
+        """Try to open an existing config file."""
+        with self._lock:
+            if os.path.exists(config_path):
+                log.info("parsing existing user interaction config file in %s", config_path)
+                with open(config_path, "r") as f:
+                    self._config.readfp(f)
+
+    def write_out_config_file(self, config_path=iutil.getSysroot() + CONFIG_FILE_PATH):
```

Comment:
Good catch! That could indeed cause issues as the value returned by getSysroot() can change during the installation.